zebra-state/src/service/check.rs

```diff
@@ -86,6 +86,41 @@ where
     Ok(())
 }
 
+#[tracing::instrument(
+    name = "contextual_validation_for_chain",
+    fields(?network),
+    skip(prepared, network)
+)]
+pub(crate) fn block_is_contextually_valid_for_chain(
+    prepared: &PreparedBlock,
+    network: Network,
+    mmr_hash: &ChainHistoryMmrRootHash,
+) -> Result<(), ValidateContextError> {
+    match prepared
+        .block
+        .commitment(network)
+        .expect("should have commitment (TODO: confirm)")
+    {
+        block::Commitment::PreSaplingReserved(_) => todo!("Confirm where this is being checked"),
+        block::Commitment::FinalSaplingRoot(_) => todo!("Confirm where this is being checked"),
+        block::Commitment::ChainHistoryActivationReserved => {
+            todo!("Confirm where this is being checked")
+        }
```

Comment:
We don't need to check these variants, because Zebra's mandatory `Canopy` activation checkpoint makes those blocks skip the non-finalized state.

It's all semantic verification, so if we ever choose to implement it, we'll do that in the `CheckpointVerifier`.

Reply:
Got it, changed it and added a comment explaining